payment/src/main/java/org/killbill/billing/payment/core/sm/PaymentAutomatonRunner.java

```diff
@@ -128,8 +129,10 @@ public PaymentStateContext buildPaymentStateContext(final boolean isApiPayment,
                                                         final Iterable<PluginProperty> properties,
                                                         final CallContext callContext,
                                                         final InternalCallContext internalCallContext) throws PaymentApiException {
+
         // Retrieve the payment id from the payment external key if needed
         final UUID effectivePaymentId = paymentId != null ? paymentId : retrievePaymentId(paymentExternalKey, paymentTransactionExternalKey, internalCallContext);
+        verifyPaymentMethodExistsForRefund(effectivePaymentId, transactionType, internalCallContext);
```

Comment:
You are probably right on the root cause - I don't understand why we call `getPaymentMethodIncludedDeleted` though (and not `getPaymentMethod`). I would try to change this and verify all tests still pass.